Src/VimCore/CoreInterfaces.fs

```diff
@@ -1802,35 +1810,42 @@ type BlockSpan =
     member x.TextBuffer =  x.Start.Snapshot.TextBuffer
 
     member private x.GetBlockSpansCore (makeSpan: ITextSnapshotLine -> int -> 'T): NonEmptyCollection<'T> =
-        let snapshot = x.Snapshot
-        let beforeSpaces = x.BeforeSpaces
-        let lineNumber = x.Start.LineNumber
-        let list = System.Collections.Generic.List<'T>()
-        for i = lineNumber to ((x._height - 1) + lineNumber) do
-            match SnapshotUtil.TryGetLine snapshot i with
-            | None -> ()
-            | Some line -> 
-                let span = makeSpan line beforeSpaces
-                list.Add(span)
-
-        list
+        let x = x
+        seq {
+            for i = 0 to x.Height - 1 do
+                let lineNumber = x.Start.LineNumber + i
+                match SnapshotUtil.TryGetLine x.Snapshot lineNumber with
+                | None -> ()
+                | Some line -> 
+                    yield makeSpan line x.BeforeSpaces
+        }
         |> NonEmptyCollectionUtil.OfSeq 
         |> Option.get
 
+    member private x.GetLastColumn line = SnapshotColumn(line, line.End)

```

Comment:
Shouldn't this be called `GetEndColumn`?

Reply:
Ooh, yes, it should.